Cpp_test/lib/ipc-shm.cpp

```diff
@@ -0,0 +1,95 @@
+#include "ipc-shm.h"
+
+IPCShm::~IPCShm()
+{
+	if (this->shm_ptr == nullptr)
```

Comment:
```suggestion
if (this->shm_ptr != nullptr)
```
You shall not munmap a nullptr.

Reply:
Yes, stupid me, do not munmap a nullptr.